From 05559aa439d8ffdfa9b8ccddecaa448ecc33dc0d Mon Sep 17 00:00:00 2001
From: Jiafei Pan <Jiafei.Pan@nxp.com>
Date: Thu, 1 Jun 2023 14:08:29 +0800
Subject: [PATCH 57/68] configs: arm64: separate imx8mm virtio_net cell

Separate virtio_net cell to be a standalone cell in order to sync
with other i.mx platforms.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>
---
 .../arm64/imx8mm-harpoon-freertos-virtio.c    | 22 +++++++++++++++++++
 configs/arm64/imx8mm-harpoon-rtos.h           | 13 +++++++++++
 2 files changed, 35 insertions(+)
 create mode 100644 configs/arm64/imx8mm-harpoon-freertos-virtio.c

diff --git a/configs/arm64/imx8mm-harpoon-freertos-virtio.c b/configs/arm64/imx8mm-harpoon-freertos-virtio.c
new file mode 100644
index 00000000..38febd3d
--- /dev/null
+++ b/configs/arm64/imx8mm-harpoon-freertos-virtio.c
@@ -0,0 +1,22 @@
+/*
+ * i.MX8MM target - FreeRTOS virtio_net config
+ *
+ * Copyright 2023 NXP
+ *
+ * Authors:
+ *  Jiafei Pan <Jiafei.Pan@nxp.com>
+ *
+ * This work is licensed under the terms of the GNU GPL, version 2.  See
+ * the COPYING file in the top-level directory.
+ */
+
+#include "cell-helper.h"
+#include "imx8mm-harpoon-rtos.h"
+
+#define CONFIG_INMATE_NAME		"freertos"
+#define CONFIG_INMATE_CORE_BITMAP	(0b1000)
+
+/* Base, memory & peripherals, GIC, IVSHMEM PCI */
+#include "imx8m-harpoon-config-virtio.h"
+
+#include "cell-create.h"
diff --git a/configs/arm64/imx8mm-harpoon-rtos.h b/configs/arm64/imx8mm-harpoon-rtos.h
index 9d7dc2e2..facfc04f 100644
--- a/configs/arm64/imx8mm-harpoon-rtos.h
+++ b/configs/arm64/imx8mm-harpoon-rtos.h
@@ -72,6 +72,9 @@ typedef enum IRQn {
 	MMIO_REGION_RW( 0x30be0000, 0x30be0000, KB(4))     /* ENET */
 
 #define RTOS_INDUS_MEMORY_REGIONS \
+	MMIO_REGION_RW( 0x30be0000, 0x30be0000, KB(4))     /* ENET */
+
+#define RTOS_VIRTIO_MEMORY_REGIONS \
 	MMIO_REGION_RW( 0x30be0000, 0x30be0000, KB(4)),     /* ENET */ \
 	MEM_REGION_RWS( 0x40000000, 0x40000000, GB(1))     /* data buffer */
 
@@ -87,6 +90,8 @@ typedef enum IRQn {
 	RTOS_AUDIO_IRQCHIP0_BITMAP1
 #define RTOS_INDUS_IRQCHIP0_BITMAP1 \
 	0
+#define RTOS_VIRTIO_IRQCHIP0_BITMAP1 \
+	0
 
 /* interrupts 64..95 */
 #define RTOS_COMMON_IRQCHIP0_BITMAP2 \
@@ -99,6 +104,8 @@ typedef enum IRQn {
 	RTOS_AUDIO_IRQCHIP0_BITMAP2
 #define RTOS_INDUS_IRQCHIP0_BITMAP2 \
 	0
+#define RTOS_VIRTIO_IRQCHIP0_BITMAP2 \
+	0
 
 /* interrupts 96..127 */
 #define RTOS_COMMON_IRQCHIP0_BITMAP3 \
@@ -110,6 +117,8 @@ typedef enum IRQn {
 	RTOS_AUDIO_IRQCHIP0_BITMAP3
 #define RTOS_INDUS_IRQCHIP0_BITMAP3 \
 	0
+#define RTOS_VIRTIO_IRQCHIP0_BITMAP3 \
+	0
 
 /* interrupts 128..159 */
 #define RTOS_COMMON_IRQCHIP0_BITMAP4 \
@@ -127,6 +136,8 @@ typedef enum IRQn {
 		IRQnMAP(ENET_MAC0_Rx_Tx_Done2_IRQn) | \
 		IRQnMAP(ENET_IRQn)                  | \
 		IRQnMAP(ENET_1588_IRQn)
+#define RTOS_VIRTIO_IRQCHIP0_BITMAP4 \
+		IRQnMAP(ENET_IRQn)
 
 /* interrupts 160..191 */
 #define RTOS_COMMON_IRQCHIP1_BITMAP1 \
@@ -138,6 +149,8 @@ typedef enum IRQn {
 	0
 #define RTOS_INDUS_IRQCHIP1_BITMAP1 \
 	0
+#define RTOS_VIRTIO_IRQCHIP1_BITMAP1 \
+	0
 
 /* IVSHMEM PCI */
 #define	RTOS_COMMON_PCI_IVSHMEM \
-- 
2.34.1

